server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java

```diff
@@ -41,18 +45,43 @@ public SessionHandler(SessionService sessionService,
         this.backend = backend;
         this.statementStatusChecker = statementStatusChecker;
         this.statusTracker = statusTracker;
-        this.appConfiguration = appConfiguration;
+        this.sessionConfiguration = appConfiguration.getSessionConfiguration();
     }
 
     public void launch(Application application, Consumer<Throwable> errorHandler) {
         var app = new SparkApp(application.getSubmitParams(), errorHandler);
         app.launch(backend.getSubmitConfiguration(application));
     }
 
+    @EventListener
+    @Async
+    public void startPermanentSession(StartupEvent event) {
+        restartPermanentSession();
+    }
+
+    private void restartPermanentSession() {
+        var sessionId = sessionConfiguration.getPermanentSessionId();
```

Comment:
I think we should make it possible to have multiple permanent sessions.

Reply:
What for?

Reply:
Jupyter, for example.

Reply:
Same permanent session could have few statements, but PythonSessionIntegration.java need some improvement to support that.

Reply:
One session should be used for one context only. i.e. different notebooks from Jupyter shouldn't use the same session because resources are allocated per session and different users should also not interfere with each other.

Reply:
Jupyter should use regular sessions with timeout handling.

Reply:
Multiple permanent sessions support will be implemented with next PR.